src/python/gudhi/tensorflow/cubical_layer.py

```diff
@@ -0,0 +1,72 @@
+import numpy               as np
+import tensorflow          as tf
+from ..cubical_complex  import CubicalComplex
+
+######################
+# Cubical filtration #
+######################
+
+# The parameters of the model are the pixel values.
+
+def _Cubical(X, dimensions):
+    # Parameters: X (image),
+    #             dimensions (homology dimensions)
+
+    # Compute the persistence pairs with Gudhi
+    Xs = X.shape
+    cc = CubicalComplex(dimensions=Xs, top_dimensional_cells=X.flatten())
+    cc.persistence()
+
+    L_cofs = []
+    for dim in dimensions:
+
+        try:
+            cof = cc.cofaces_of_persistence_pairs()[0][dim]
+        except IndexError:
+            cof = np.array([])
+
+        # Retrieve and ouput image indices/pixels corresponding to positive and negative simplices
+        D = len(Xs) if len(cof) > 0 else 1
+        ocof = np.array([0 for _ in range(D*2*cof.shape[0])])
+        count = 0
+        for idx in range(0,2*cof.shape[0],2):
+            ocof[D*idx:D*(idx+1)]     = np.unravel_index(cof[count,0], Xs)
+            ocof[D*(idx+1):D*(idx+2)] = np.unravel_index(cof[count,1], Xs)
+            count += 1
+        L_cofs.append(np.array(ocof, dtype=np.int32))
+
+    return L_cofs
+
+class CubicalLayer(tf.keras.layers.Layer):
+    """
+    TensorFlow layer for computing cubical persistence out of a cubical complex
+    """
+    def __init__(self, dimensions=[1], **kwargs):
```

Comment:
Is there a reason to believe that dimension 1 is a good default?

Reply:
No reason ;-) I changed to 0.

Reply:
Actually, I was more thinking of **not** having a default (same for the other layers), so the user has to think at list 1 second what dimension they want, and so it is clearer when reading the code what dimension is used. Unless there is a good reason to have a default? For cubical specifically, [0, ..., dim-1] could be a sensible default, but that would encourage users not to prune that list when they don't use everything.

Reply:
I agree with you. I've seen a few times people applying differentiation blindly so it makes sense to ask explicitly for dimensions. I modified the code accordingly.